src/main/java/org/folio/service/caches/JobProfileSnapshotCache.java

```diff
@@ -0,0 +1,72 @@
+package org.folio.service.caches;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.Json;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.jaxrs.model.ProfileSnapshotWrapper;
+import org.folio.rest.util.OkapiConnectionParams;
+import org.folio.rest.util.RestUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.apache.http.HttpStatus;
+
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class JobProfileSnapshotCache {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  @Value("${orders.profile-snapshot-cache.expiration.time.seconds:3600}")
+  private long cacheExpirationTime;
+  private AsyncCache<String, Optional<ProfileSnapshotWrapper>> cache;
+
+  @Autowired
+  public JobProfileSnapshotCache(Vertx vertx) {
+    cache = Caffeine.newBuilder()
+      .expireAfterAccess(cacheExpirationTime, TimeUnit.SECONDS)
+      .executor(task -> vertx.runOnContext(v -> task.run()))
+      .buildAsync();
+  }
+
+  public Future<Optional<ProfileSnapshotWrapper>> get(String profileSnapshotId, OkapiConnectionParams params) {
+    try {
+      return Future.fromCompletionStage(cache.get(profileSnapshotId, (key, executor) -> loadJobProfileSnapshot(key, params)));
+    } catch (Exception e) {
+      LOGGER.warn("Error loading ProfileSnapshotWrapper by id: '{}'", profileSnapshotId, e);
+      return Future.failedFuture(e);
+    }
+  }
+
+  private CompletableFuture<Optional<ProfileSnapshotWrapper>> loadJobProfileSnapshot(String profileSnapshotId, OkapiConnectionParams params) {
+    LOGGER.debug("Trying to load jobProfileSnapshot by id  '{}' for cache, okapi url: {}, tenantId: {}", profileSnapshotId, params.getOkapiUrl(), params.getTenantId());
+
+    return RestUtil.doRequest(params, "/data-import-profiles/jobProfileSnapshots/" + profileSnapshotId, HttpMethod.GET, null)
```

Comment:
Is using RestUtil the standard pattern in mod-orders?

Reply:
this is the standard pattern for Folio. I think we can use it.